tests/block-creation.spec.js

```diff
@@ -0,0 +1,60 @@
+import {expect, test} from "@playwright/test";
+import {HeynotePage} from "./test-utils.js";
+
+let heynotePage
+
+test.beforeEach(async ({ page }) => {
+    console.log("beforeEach")
+    heynotePage = new HeynotePage(page)
+    await heynotePage.goto()
+
+    expect((await heynotePage.getBlocks()).length).toBe(1)
+    heynotePage.setContent(`
+∞∞∞text
+Block A
+∞∞∞text
+Block B
+∞∞∞text
+Block C`)
+    // check that blocks are created
+    expect((await heynotePage.getBlocks()).length).toBe(3)
+
+    // check that visual block layers are created
+    expect(await page.locator("css=.heynote-blocks-layer > div").count()).toBe(3)
+
+    // select the second block
+    await page.locator("body").press("ArrowUp")
+});
+
+test("create block before current", async ({ page }) => {
+    await runTest(page, "Alt+Enter", ['A', 'D', 'B', 'C'])
+})
+
+test("create block after current", async ({ page }) => {
+    await runTest(page, heynotePage.isMac ? "Meta+Enter" : "Control+Enter",['A', 'B', 'D', 'C'])
+})
+
+test("create block before first", async ({ page }) => {
+    await runTest(page, "Alt+Shift+Enter",['D', 'A', 'B', 'C'])
+})
+
+test("create block after last", async ({ page }) => {
+    await runTest(page, heynotePage.isMac ? "Meta+Shift+Enter" : "Control+Shift+Enter",['A', 'B', 'C', 'D'])
+})
```

Comment:
Nice with all these test cases!

The Meta/Control difference would be nice to get out of the test code I think. Something like:

```javascript
const getCommand = (keys) => {
    let modKey = heynotePage.isMac ? "Meta" : "Control";
    return keys.replaceAll("Mod", modKey);
}
```

To be used like `getCommand("Mod+Shift+Enter")` in the test code.

Reply:
I'll have a look at that